Add Yoto integration documentation (#45446)
home-assistant/home-assistant.github.io: source/_integrations/yoto.markdown

---
title: Yoto
description: Instructions on how to integrate Yoto players with Home Assistant.
ha_category:
  - Media Player
ha_iot_class: Cloud Push

> ha_release: '2026.6'

Reviewer: ```suggestion
ha_release: 2026.6
```

For more details, see the [Yoto Developers documentation](https://yoto.dev/get-started/start-here/).

{% enddetails %}

{% include integrations/config_flow.md %}

During setup, Home Assistant asks for your Yoto **Client ID** and **Client secret** through the [Application Credentials](/integrations/application_credentials/) flow, then opens the Yoto authorization page so you can grant access. After you approve, Home Assistant creates one {% term device %} and one media player {% term entity %} for every Yoto player in your family.

## Supported functionality

The integration provides one media player entity per Yoto player. Each entity supports:

- Play, pause, and stop
- Skip to the previous or next track
- Seek to a position within the current track
- Set the volume directly, or step up and down in the 16 hardware steps the player uses
- Show the currently playing track title, the card title and author, and the card cover art as media artwork
- Show the player as _off_ when it is asleep or disconnected from the Yoto cloud

Yoto players cannot be powered on remotely. Home Assistant reports the player as _off_ when it is offline but cannot wake it up.

### Play media

> When you use the [`media_player.play_media`](/integrations/media_player/#action-media_playerplay_media) action, pass the Yoto identifier in `media_content_id`. The action also requires `media_content_type`.

Reviewer: ```suggestion
When you use the [`media_player.play_media`](/integrations/media_player/#action-media_playerplay_media) action, pass the Yoto card identifier in `media_content_id` and set `media_content_type` to `music`. The action also requires `media_content_type`.
```

The `media_content_id` value supports two formats:

- A bare card ID. Playback starts from the beginning of the card.
- A structured ID in the form `<card_id>+<chapter_key>+<track_key>+<seconds_in>`. Each segment after the card ID is optional. Leave a segment empty to keep its default.

## Data updates

> The integration receives real-time playback updates from each Yoto player over MQTT. To keep the reported status fresh even when the player has not changed state, the integration also asks every player to push a status snapshot every 60 seconds.

Reviewer: ```suggestion
The integration receives real-time playback updates from each Yoto player over MQTT. To keep the reported status fresh even when the player has not changed state, the integration also requests a status snapshot from each player every 60 seconds.
```
